src/Nest/CommonOptions/TimeUnit/Time.cs

```diff
@@ -20,7 +21,7 @@ public class Time : IComparable<Time>, IEquatable<Time>
 		private const double MicrosecondsInAMillisecond = 10;
 
 		private static readonly Regex ExpressionRegex =
-			new Regex(@"^(?<factor>[-+]?\d+(?:\.\d+)?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
+			new Regex(@"^(?<factor>[-+]?[\d\.e\-]+?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
```

Comment:
> `(?:\d{1,})(?:\.(?:\d+[e][+\-]?)?\d+)?)` 

always requires the mantissa to include a decimal value e.g.

```
6.0e5
```
but does not match 
```
6e5
```
> doing insensitive matching only e should be fine and now with this we do not match on 123e123123 or 123.e1231

Both `123e123123` or `123.e1231` are valid notation

> `\s* => [ ]{0,10}`

Agreed, although maybe as `\s{0,10}`?

----

How about

```
^(?<factor>[+\-]?(?<mantissa>(?:\d+(\.\d*)?)|(?:\.\d+))?(?(mantissa)[e](?<exponent>[+\-]?\d+))?)\s{0,10}(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$
```
matches all of

```
1
+1
-1
1y
.1
1.2
1.2w
1E2
.1e2
1e-2
1.e2
1.e-2
1.2e3
1.2e-3
1.2e3d
1.2e-3h
```
and requires a number before the decimal point, after the decimal point or both. So things like

```
.e1
```
doesn't match

and also a mantissa must be present when exponential notation is used, so

```
e1
```

doesn't match